profile/ucs_chassis_profile/terraform/ucs_chassis_profile.tf

```diff
@@ -1,16 +1,163 @@
+  snmp_users {
+    name             = "demouser"
+    privacy_type     = "AES"
+    auth_password    = var.auth_password
+    privacy_password = var.privacy_password
+    security_level   = "AuthPriv"
+    auth_type        = "SHA"
+    object_type      = "snmp.User"
+  }
+  snmp_traps {
+    destination = "10.10.10.1"
+    enabled     = false
+    port        = 660
+    type        = "Trap"
+    user        = "demouser"
+    nr_version  = "V3"
+    object_type = "snmp.Trap"
+  }
+  profiles {
+    moid        = var.profile
+    object_type = "server.Profile"
+  }
+  organization {
+    object_type = "organization.Organization"
+    moid = data.intersight_organization_organization.organization.id
+  }
+}
+
+resource "intersight_thermal_policy" "sample_thermal_policy" {
+  name         = "sample_thermal_policy"
+  description  = "Sample thermal policy"
+  organization {
+    object_type = "organization.Organization"
+    moid = data.intersight_organization_organization.organization.id
+  }
+}
+
+resource "intersight_power_policy" "sample_power_policy" {
+  name         = "sample_power_policy"
+  description  = "Sample power policy"
+  organization {
+    object_type = "organization.Organization"
+    moid = data.intersight_organization_organization.organization.id
+  }
+}
+
+resource "intersight_chassis_profile" "sample_chassis_profile" {
   name            = "chassis_profile1"
   description     = "chassis profile"
   type            = "instance"
   target_platform = "FIAttached"
   action          = "Validate"
-  config_context {
-    object_type    = "policy.ConfigContext"
-    control_action = "deploy"
-    error_state    = "Validation-error"
-  }
   organization {
     object_type = "organization.Organization"
-    moid        = var.organization
+    moid = data.intersight_organization_organization.organization.id
   }
+
+  policy_bucket = [
+    {
+      object_type = "access.Policy"
+      moid        = intersight_access_policy.access1.moid
+      class_id    = "policy.AbstractPolicy"
+      additional_properties = ""
+      selector = ""
+    },
+    {
+      object_type = "snmp.Policy"
+      moid        = intersight_snmp_policy.snmp1.moid
+      class_id    = "policy.AbstractPolicy"
+      additional_properties = ""
+      selector = ""
+    },
+    {
+      object_type = "thermal.Policy"
+      moid        = intersight_thermal_policy.sample_thermal_policy.moid
+      class_id    = "policy.AbstractPolicy"
+      additional_properties = ""
+      selector = ""
+    },
+    {
+      object_type = "power.Policy"
+      moid        = intersight_power_policy.sample_power_policy.moid
+      class_id    = "policy.AbstractPolicy"
+      additional_properties = ""
+      selector = ""
+    },
+  ]
+}
+
+variable "auth_password" {
+  type        = string
+  description = "value for auth_password"
+}
+
+variable "privacy_password" {
+  type        = string
+  description = "value for privacy password"
+}
+
+variable "profile" {
```

Comment:
why we are declaring variable for profile

Reply:
Removed .